lib/backend/grpcclient/grpc.go

```
@@ -0,0 +1,123 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcclient
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type Backend struct {
+	clock clockwork.Clock
+}
+
+// Config represents JSON config for grpc backend
+type Config struct {
+	// TODO(michaelmcallister): implementation
+}
+
+// GetName returns the name of the backend as it appears in 'storage/type' section
+// in Teleport YAML file. This function is a part of backend API
+func GetName() string {
+	return "grpc"
+}
+
+// keep this here to test interface conformance
+var _ backend.Backend = &Backend{}
+
+// New returns new instance of grpc-powered backend
+func New(ctx context.Context, params backend.Params) (*Backend, error) {
```

Maybe we should return an error here until the full implementation is included?